Update/file renaming

hamshkhawar opened the pull request:
Old Pull request https://github.com/PolusAI/polus-plugins/pull/210

Reopened the PR for file renaming plugin as the original branch got deleted and this plugin wasn't merged to the master branch
Updated this plugin to new plugin standards

camilovelezr commented:
This looks good. Only one thing when running pytest:
```
test_main.py::test_invalid_input_raises_error[poly0] XPASS
test_main.py::test_extract_named_grp_matches_bad_pattern_invalid_input_fails XPASS 
```
If I understand this correctly, these two tests should fail but they are actually not failing. Could you confirm this?

hamshkhawar commented:
> This looks good. Only one thing when running pytest:
> 
> ```
> test_main.py::test_invalid_input_raises_error[poly0] XPASS
> test_main.py::test_extract_named_grp_matches_bad_pattern_invalid_input_fails XPASS 
> ```
> 
> If I understand this correctly, these two tests should fail but they are actually not failing. Could you confirm this?

@camilovelezr I used @pytest.mark.xfail marker for this those function where we are certain that this function is going to fail. pytest will execute those xfailed tests but it will not be considered as part failed or passed tests and it is reported as 2 xfailed  tests and 21 passed test.

camilovelezr commented:
@hamshkhawar I don't get those same results when running pytest:
```
(.venv) (base) ➜  tests git:(update/file-renaming) ✗ pytest -v
=========================================================== test session starts ============================================================
platform darwin -- Python 3.10.5, pytest-7.3.1, pluggy-1.0.0 -- /polus-plugins/formats/file-renaming-plugin/.venv/bin/python
cachedir: .pytest_cache
rootdir: /polus-plugins
collected 25 items                                                                                                                         

test_main.py::test_duplicate_channels_to_digit[poly0] PASSED                                                                         [  4%]
test_main.py::test_duplicate_channels_to_digit_non_spec_digit_len[poly0] PASSED                                                      [  8%]
test_main.py::test_invalid_input_raises_error[poly0] XPASS                                                                           [ 12%]
test_main.py::test_non_alphanum_inputs_percentage_sign[poly0] PASSED                                                                 [ 16%]
test_main.py::test_numeric_fixed_width[poly0] PASSED                                                                                 [ 20%]
test_main.py::test_alphanumeric_fixed_width[poly0] PASSED                                                                            [ 24%]
test_main.py::test_alphanumeric_variable_width[poly0] PASSED                                                                         [ 28%]
test_main.py::test_parenthesis[poly0] PASSED                                                                                         [ 32%]
test_main.py::test_two_chan_to_digit[poly0] PASSED                                                                                   [ 36%]
test_main.py::test_three_chan_to_digit[poly0] PASSED                                                                                 [ 40%]
test_main.py::test_three_char_chan[poly0] PASSED                                                                                     [ 44%]
test_main.py::test_varied_digits[poly0] PASSED                                                                                       [ 48%]
test_main.py::test_spaces[poly0] PASSED                                                                                              [ 52%]
test_main.py::test_non_alphanum_float[poly0] PASSED                                                                                  [ 56%]
test_main.py::test_dashes_parentheses[poly0] PASSED                                                                                  [ 60%]
test_main.py::test_map_pattern_grps_to_regex_valid_input PASSED                                                                      [ 64%]
test_main.py::test_convert_to_regex_valid_input PASSED                                                                               [ 68%]
test_main.py::test_specify_len_valid_input PASSED                                                                                    [ 72%]
test_main.py::test_get_char_to_digit_grps_returns_unique_keys_valid_input PASSED                                                     [ 76%]
test_main.py::test_extract_named_grp_matches_valid_input PASSED                                                                      [ 80%]
test_main.py::test_extract_named_grp_matches_bad_pattern_invalid_input_fails XPASS                                                   [ 84%]
test_main.py::test_extract_named_grp_matches_duplicate_namedgrp_invalid_input XFAIL                                                  [ 88%]
test_main.py::test_str_to_int_valid_input PASSED                                                                                     [ 92%]
test_main.py::test_letters_to_int_returns_cat_index_dict_valid_input PASSED                                                          [ 96%]
test_main.py::test_letters_to_int_returns_error_invalid_input XFAIL                                                                  [100%]

================================================= 21 passed, 2 xfailed, 2 xpassed in 0.14s =================================================
```

hamshkhawar commented:
> @hamshkhawar I don't get those same results when running pytest:
> 
> ```
> (.venv) (base) ➜  tests git:(update/file-renaming) ✗ pytest -v
> =========================================================== test session starts ============================================================
> platform darwin -- Python 3.10.5, pytest-7.3.1, pluggy-1.0.0 -- /polus-plugins/formats/file-renaming-plugin/.venv/bin/python
> cachedir: .pytest_cache
> rootdir: /polus-plugins
> collected 25 items                                                                                                                         
> 
> test_main.py::test_duplicate_channels_to_digit[poly0] PASSED                                                                         [  4%]
> test_main.py::test_duplicate_channels_to_digit_non_spec_digit_len[poly0] PASSED                                                      [  8%]
> test_main.py::test_invalid_input_raises_error[poly0] XPASS                                                                           [ 12%]
> test_main.py::test_non_alphanum_inputs_percentage_sign[poly0] PASSED                                                                 [ 16%]
> test_main.py::test_numeric_fixed_width[poly0] PASSED                                                                                 [ 20%]
> test_main.py::test_alphanumeric_fixed_width[poly0] PASSED                                                                            [ 24%]
> test_main.py::test_alphanumeric_variable_width[poly0] PASSED                                                                         [ 28%]
> test_main.py::test_parenthesis[poly0] PASSED                                                                                         [ 32%]
> test_main.py::test_two_chan_to_digit[poly0] PASSED                                                                                   [ 36%]
> test_main.py::test_three_chan_to_digit[poly0] PASSED                                                                                 [ 40%]
> test_main.py::test_three_char_chan[poly0] PASSED                                                                                     [ 44%]
> test_main.py::test_varied_digits[poly0] PASSED                                                                                       [ 48%]
> test_main.py::test_spaces[poly0] PASSED                                                                                              [ 52%]
> test_main.py::test_non_alphanum_float[poly0] PASSED                                                                                  [ 56%]
> test_main.py::test_dashes_parentheses[poly0] PASSED                                                                                  [ 60%]
> test_main.py::test_map_pattern_grps_to_regex_valid_input PASSED                                                                      [ 64%]
> test_main.py::test_convert_to_regex_valid_input PASSED                                                                               [ 68%]
> test_main.py::test_specify_len_valid_input PASSED                                                                                    [ 72%]
> test_main.py::test_get_char_to_digit_grps_returns_unique_keys_valid_input PASSED                                                     [ 76%]
> test_main.py::test_extract_named_grp_matches_valid_input PASSED                                                                      [ 80%]
> test_main.py::test_extract_named_grp_matches_bad_pattern_invalid_input_fails XPASS                                                   [ 84%]
> test_main.py::test_extract_named_grp_matches_duplicate_namedgrp_invalid_input XFAIL                                                  [ 88%]
> test_main.py::test_str_to_int_valid_input PASSED                                                                                     [ 92%]
> test_main.py::test_letters_to_int_returns_cat_index_dict_valid_input PASSED                                                          [ 96%]
> test_main.py::test_letters_to_int_returns_error_invalid_input XFAIL                                                                  [100%]
> 
> ================================================= 21 passed, 2 xfailed, 2 xpassed in 0.14s =================================================
> ```

I fixed that there were two functions were passing but I had @pytest.mark.xfail marker on it. I removed it and now it shows only 23 passed, 2 xfailed in 0.35s
